### PR TITLE
PP-13234: Adding a pipeline for run-amock

### PR DESCRIPTION
**Pull request by nataliecarey**
Two things:

1. I don't know how to run this to check it, I've made the changes I'd expect to work but haven't run them yet (reaching out to the relevant people to go through that process with).
2. I've added `short_name` and `really_short_name` that should probably be renamed if they're needed.  As I'm not sure they're needed I haven't put much effort into naming them yet.

### CODE REVIEW

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

File: ci/pkl-pipelines/pay-deploy/pay-js.pkl [hunk 1]
```diff
@@ -11,11 +11,21 @@ local class PayJsLibraryPipeline {
 
 local payJsLibraries: Listing<PayJsLibraryPipeline> = new {
   new {
-    name = "commons"
+    name = "pay-js-commons"
```
Comment (jfharden) [inline ci/pkl-pipelines/pay-deploy/pay-js.pkl:14]: These are instances of the classes on line 7, so you need to add the new properties to that class.

One of the github actions is failing, if you click on the link for that it should take you to the concourse job that failed, opening the failed task will show you the error.

You can also run it yourself and see them locally.


If you `brew install pkl` you can then run in `pay-ci/ci/pkl-pipelines`

```
pkl eval pay-deploy/pay-js.pkl
```

Comment (nataliecarey) [reply ci/pkl-pipelines/pay-deploy/pay-js.pkl:14]: Thanks, I'll check that out :)

**Comment by alphagov-pay-ci-concourse** [comment]
# Changes for pay-deploy/pay-js.pkl

<details><summary>Diff of YAML generated from pay-deploy/pay-js.pkl</summary>

```diff
--- pay-ci/ci/pkl-pipelines/pay-deploy/pay-js.yml
+++ pkl-pipeline-pr/ci/pkl-pipelines/pay-deploy/pay-js.yml
@@ -51,6 +51,20 @@
       GITHUB_TOKEN: ((github-access-token))
     input_mapping:
       src: js-metrics-git-release
+- name: version-and-push-run-amock
+  plan:
+  - in_parallel:
+    - get: run-amock-git-release
+      trigger: true
+    - get: pay-ci
+  - task: npm-version-and-create-pr
+    file: pay-ci/ci/tasks/npm-version-and-create-pr.yml
+    params:
+      BASE: main
+      REPO: alphagov/run-amock
+      GITHUB_TOKEN: ((github-access-token))
+    input_mapping:
+      src: run-amock-git-release
 resources:
 - name: pipeline-source
   type: git
@@ -71,7 +85,7 @@
     username: alphagov-pay-ci-concourse
     password: ((github-access-token))
   icon: github
-- name: js-commons-git-release
+- name: js-pay-js-commons-git-release
   type: git
   source:
     uri: https://github.com/alphagov/pay-js-commons
@@ -82,7 +96,7 @@
       exclude:
       - \[automated release\]
   icon: github
-- name: js-metrics-git-release
+- name: js-pay-js-metrics-git-release
   type: git
   source:
     uri: https://github.com/alphagov/pay-js-metrics
@@ -93,6 +107,17 @@
       exclude:
       - \[automated release\]
   icon: github
+- name: js-run-amock-git-release
+  type: git
+  source:
+    uri: https://github.com/alphagov/run-amock
+    branch: main
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    commit_filter:
+      exclude:
+      - \[automated release\]
+  icon: github
 groups:
 - name: js-commons
   jobs:
@@ -100,6 +125,9 @@
 - name: js-metrics
   jobs:
   - version-and-push-metrics
+- name: run-amock
+  jobs:
+  - version-and-push-run-amock
 - name: update-pipeline
   jobs:
   - update-pipeline
```
</details>


<details><summary>Concourse set-pipeline dry-run for `pay-deploy/pay-js.pkl`</summary>

```diff
groups:
  group update-pipeline has changed:
  jobs:
  - update-pipeline
  name: update-pipeline
  
  group run-amock has been added:
+ jobs:
+ - version-and-push-run-amock
+ name: run-amock
  
resources:
  resource js-commons-git-release has been removed:
- icon: github
- name: js-commons-git-release
- source:
-   branch: master
-   commit_filter:
-     exclude:
-     - \[automated release\]
-   password: ((github-access-token))
-   uri: https://github.com/alphagov/pay-js-commons
-   username: alphagov-pay-ci-concourse
- type: git
  
  resource js-metrics-git-release has been removed:
- icon: github
- name: js-metrics-git-release
- source:
-   branch: main
-   commit_filter:
-     exclude:
-     - \[automated release\]
-   password: ((github-access-token))
-   uri: https://github.com/alphagov/pay-js-metrics
-   username: alphagov-pay-ci-concourse
- type: git
  
  resource js-pay-js-commons-git-release has been added:
+ icon: github
+ name: js-pay-js-commons-git-release
+ source:
+   branch: master
+   commit_filter:
+     exclude:
+     - \[automated release\]
+   password: ((github-access-token))
+   uri: https://github.com/alphagov/pay-js-commons
+   username: alphagov-pay-ci-concourse
+ type: git
  
  resource js-pay-js-metrics-git-release has been added:
+ icon: github
+ name: js-pay-js-metrics-git-release
+ source:
+   branch: main
+   commit_filter:
+     exclude:
+     - \[automated release\]
+   password: ((github-access-token))
+   uri: https://github.com/alphagov/pay-js-metrics
+   username: alphagov-pay-ci-concourse
+ type: git
  
  resource js-run-amock-git-release has been added:
+ icon: github
+ name: js-run-amock-git-release
+ source:
+   branch: main
+   commit_filter:
+     exclude:
+     - \[automated release\]
+   password: ((github-access-token))
+   uri: https://github.com/alphagov/run-amock
+   username: alphagov-pay-ci-concourse
+ type: git
  
jobs:
  job version-and-push-run-amock has been added:
+ name: version-and-push-run-amock
+ plan:
+ - in_parallel:
+     steps:
+     - get: run-amock-git-release
+       trigger: true
+     - get: pay-ci
+ - file: pay-ci/ci/tasks/npm-version-and-create-pr.yml
+   input_mapping:
+     src: run-amock-git-release
+   params:
+     BASE: main
+     GITHUB_TOKEN: ((github-access-token))
+     REPO: alphagov/run-amock
+   task: npm-version-and-create-pr
  
pipeline name: pay-js

Dry-run mode was set, exiting.
```

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/bastion.yml
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/node-runner.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
# Changes for pay-deploy/pay-js.pkl

<details><summary>Diff of YAML generated from pay-deploy/pay-js.pkl</summary>

```diff
--- pay-ci/ci/pkl-pipelines/pay-deploy/pay-js.yml
+++ pkl-pipeline-pr/ci/pkl-pipelines/pay-deploy/pay-js.yml
@@ -51,6 +51,20 @@
       GITHUB_TOKEN: ((github-access-token))
     input_mapping:
       src: js-metrics-git-release
+- name: version-and-push-run-amock
+  plan:
+  - in_parallel:
+    - get: run-amock-git-release
+      trigger: true
+    - get: pay-ci
+  - task: npm-version-and-create-pr
+    file: pay-ci/ci/tasks/npm-version-and-create-pr.yml
+    params:
+      BASE: main
+      REPO: alphagov/run-amock
+      GITHUB_TOKEN: ((github-access-token))
+    input_mapping:
+      src: run-amock-git-release
 resources:
 - name: pipeline-source
   type: git
@@ -93,6 +107,17 @@
       exclude:
       - \[automated release\]
   icon: github
+- name: run-amock-git-release
+  type: git
+  source:
+    uri: https://github.com/alphagov/run-amock
+    branch: main
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    commit_filter:
+      exclude:
+      - \[automated release\]
+  icon: github
 groups:
 - name: js-commons
   jobs:
@@ -100,6 +125,9 @@
 - name: js-metrics
   jobs:
   - version-and-push-metrics
+- name: run-amock
+  jobs:
+  - version-and-push-run-amock
 - name: update-pipeline
   jobs:
   - update-pipeline
```
</details>


<details><summary>Concourse set-pipeline dry-run for `pay-deploy/pay-js.pkl`</summary>

```diff
groups:
  group update-pipeline has changed:
  jobs:
  - update-pipeline
  name: update-pipeline
  
  group run-amock has been added:
+ jobs:
+ - version-and-push-run-amock
+ name: run-amock
  
resources:
  resource run-amock-git-release has been added:
+ icon: github
+ name: run-amock-git-release
+ source:
+   branch: main
+   commit_filter:
+     exclude:
+     - \[automated release\]
+   password: ((github-access-token))
+   uri: https://github.com/alphagov/run-amock
+   username: alphagov-pay-ci-concourse
+ type: git
  
jobs:
  job version-and-push-run-amock has been added:
+ name: version-and-push-run-amock
+ plan:
+ - in_parallel:
+     steps:
+     - get: run-amock-git-release
+       trigger: true
+     - get: pay-ci
+ - file: pay-ci/ci/tasks/npm-version-and-create-pr.yml
+   input_mapping:
+     src: run-amock-git-release
+   params:
+     BASE: main
+     GITHUB_TOKEN: ((github-access-token))
+     REPO: alphagov/run-amock
+   task: npm-version-and-create-pr
  
pipeline name: pay-js

Dry-run mode was set, exiting.
```

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/bastion.yml
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/node-runner.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

**Review by jfharden — APPROVED** [review]
Nice work 🥳

File: ci/pkl-pipelines/pay-deploy/pay-js.pkl (hunk 1 above)
Comment (nataliecarey) [reply ci/pkl-pipelines/pay-deploy/pay-js.pkl:16]: @jfharden (and anyone else who's got an opinion)

Are there any sensible semantic names for the different variations of the name?

Before this change the name was just either `commons` or `metrics` which is extended out automatically to `js-commons` and `pay-js-commons` but that's not appropriate for `run-amock`.

I'd be tempted to call them:

- `full_name` (e.g. `pay-js-commons`, `pay-js-metrics`, `run-amock`)
- `short_name` (e.g. `commons`, `metrics`, `run-amock`)
- `internal_name` (e.g. `js-commons`, `js-metrics`, `run-amock`)

Does that sound vaguely sensible?  Any other/better/different suggestions?

**Pull request merged**